### PR TITLE
Tensorflow supports inverse problems

### PR DESCRIPTION
**Pull request by ZongrenZou**

### CODE REVIEW

File: deepxde/model.py [hunk 1]
```diff
@@ -54,6 +56,7 @@ def compile(
         metrics=None,
         decay=None,
         loss_weights=None,
+        external_trainable_variables=None,
```
Comment (lululxvi) [inline deepxde/model.py:59]: Add the document for external_trainable_variables below

File: deepxde/model.py [hunk 2]
```diff
@@ -44,6 +44,8 @@ def __init__(self, data, net):
         if backend_name == "tensorflow.compat.v1":
             self.sess = None
             self.saver = None
+        elif backend_name == "tensorflow":
+            self.external_trainable_variables = None
```
Comment (lululxvi) [inline deepxde/model.py:48]: Put
```
self.external_trainable_variables = None
```
after `self.metrics` as the common variables

File: deepxde/model.py [hunk 3]
```diff
@@ -75,6 +77,10 @@ def compile(
                 weight the loss contributions. The loss value that will be minimized by
                 the model will then be the weighted sum of all individual losses,
                 weighted by the loss_weights coefficients.
+            external_trainable_variables: A Variable object or a list of Variable objects.
```
Comment (lululxvi) [inline deepxde/model.py:80]: Use: ``tf.Variable`` with the quote

File: deepxde/model.py [hunk 4]
```diff
@@ -137,13 +143,28 @@ def train_step(data_id, inputs, targets):
                 with tf.GradientTape() as tape:
                     _, losses = outputs_losses(data_id, inputs, targets)
                     total_loss = tf.math.reduce_sum(losses)
-                grads = tape.gradient(total_loss, self.net.trainable_variables)
-                opt.apply_gradients(zip(grads, self.net.trainable_variables))
+                grads = tape.gradient(
```
Comment (lululxvi) [inline deepxde/model.py:146]: trainable_variables = self.net.trainable_variables + self.external_trainable_variables

File: deepxde/model.py [hunk 5]
```diff
@@ -137,13 +143,30 @@ def train_step(data_id, inputs, targets):
...
                     _, losses = outputs_losses(data_id, inputs, targets)
                     total_loss = tf.math.reduce_sum(losses)
-                grads = tape.gradient(total_loss, self.net.trainable_variables)
-                opt.apply_gradients(zip(grads, self.net.trainable_variables))
+                trainable_variables = (
+                    self.net.trainable_variables + self.external_trainable_variables
+                )
+                grads = tape.gradient(
+                    total_loss,
+                    trainable_variables,
```
Comment (lululxvi) [inline deepxde/model.py:151]: grads = tape.gradient(total_loss, trainable_variables)

File: deepxde/model.py [hunk 6]
```diff
@@ -137,13 +143,30 @@ def train_step(data_id, inputs, targets):
...
+                )
+                opt.apply_gradients(
+                    zip(
+                        grads,
+                        trainable_variables,
```
Comment (lululxvi) [inline deepxde/model.py:156]: Remove the last ',', then black

File: deepxde/model.py [hunk 7]
```diff
@@ -32,6 +32,7 @@ def __init__(self, data, net):
         self.batch_size = None
         self.callbacks = None
         self.metrics = None
+        self.external_trainable_variables = None
```
Comment (lululxvi) [inline deepxde/model.py:35]: self.external_trainable_variables = []

File: deepxde/model.py [hunk 8]
```diff
@@ -137,13 +143,22 @@ def train_step(data_id, inputs, targets):
...
             # Callables
             self.losses = compute_losses
             self.outputs_losses = outputs_losses
             self.train_step = train_step
+            if external_trainable_variables is None:
+                self.external_trainable_variables = []
+            elif isinstance(external_trainable_variables, list):
+                self.external_trainable_variables = external_trainable_variables
+            else:
+                self.external_trainable_variables = [external_trainable_variables]
```
Comment (lululxvi) [inline deepxde/model.py:159]:
```python
if external_trainable_variables is not None:
    if not isinstance(external_trainable_variables, list):
        external_trainable_variables = [external_trainable_variables]
    self.external_trainable_variables = external_trainable_variables
```
Put this after `self.opt_name = optimizer`

File: deepxde/model.py [hunk 9]
```diff
@@ -86,6 +92,10 @@ def compile(
                 self.saver = tf.train.Saver(max_to_keep=None)
 
         self.opt_name = optimizer
+        if external_trainable_variables is not None:
+            if not isinstance(external_trainable_variables, list):
+                self.external_trainable_variables = [external_trainable_variables]
+            self.external_trainable_variables = external_trainable_variables
```
Comment (lululxvi) [inline deepxde/model.py:98]: else:
    self.external_trainable_variables = []
